Lib9c/Action/ClaimItems.cs

```diff
@@ -74,58 +94,92 @@ public override IAccount Execute(IActionContext context)
             {
                 var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
                 var inventory = states.GetInventory(inventoryAddress)
-                                ?? throw new FailedLoadStateException(
-                                    ActionTypeText,
-                                    GetSignerAndOtherAddressesHex(context, inventoryAddress),
-                                    typeof(Inventory),
-                                    inventoryAddress);
+                            ?? throw new FailedLoadStateException(
+                                ActionTypeText,
+                                GetSignerAndOtherAddressesHex(context, inventoryAddress),
+                                typeof(Inventory),
+                                inventoryAddress);
+                if (!states.TryGetState(avatarAddress, out Dictionary avatarDict))
+                {
+                    throw new FailedLoadStateException(avatarAddress, typeof(AvatarState));
+                }
 
+                var agentAddress = avatarDict[AgentAddressKey].ToAddress();
+                var favs = new List<FungibleAssetValue>();
+                var items = new List<(int id, int count)>();
                 foreach (var fungibleAssetValue in fungibleAssetValues)
                 {
-                    if (fungibleAssetValue.Currency.DecimalPlaces != 0)
+                    var wrappedTicker = fungibleAssetValue.Currency.Ticker;
+                    var separator = wrappedTicker.Contains("FAV") ? "__" : "_";
+                    var parsedTicker = wrappedTicker.Split(separator);
+                    if (parsedTicker.Length == 2)
                     {
-                        throw new ArgumentException(
-                            $"DecimalPlaces of fungibleAssetValue for claimItems are not 0: {fungibleAssetValue.Currency.Ticker}");
+                        var ticker = parsedTicker[1];
+                        var currency = Currencies.GetMinterlessCurrency(ticker);
+                        var recipientAddress =
+                            currency.DecimalPlaces > 0 ? agentAddress : avatarAddress;
```

Comment:
Note: Maybe there is a promise to use `decimalPlaces` as 0. (e.g., `rune_`, `runestone_`, `soulstone_`)